### PR TITLE
[SystemZ][z/OS] Add guard for dl_info and dladdr

### PR DESCRIPTION
**Pull request by abhina-sree**
This patch fixes the following build error on z/OS `error: unknown type name 'Dl_info'` by adding a guard to check if we have dladdr.

### CODE REVIEW

File: clang/tools/libclang/CIndexer.cpp [hunk 1]
```diff
@@ -125,21 +126,28 @@ const std::string &CIndexer::getClangResourcesPath() {
 #elif defined(_AIX)
   getClangResourcesPathImplAIX(LibClangPath);
 #else
+  bool pathFound = false;
```
Comment (fanbo-meng) [inline clang/tools/libclang/CIndexer.cpp:129]: nit: Variables should start with upper case

**Review by fanbo-meng — APPROVED** [review]
LGTM with nit

File: clang/tools/libclang/CIndexer.cpp (hunk 1 above)
Comment (abhina-sree) [reply clang/tools/libclang/CIndexer.cpp]: thanks, done!

File: clang/tools/libclang/CIndexer.cpp [hunk 2]
```diff
@@ -17,6 +17,7 @@
 #include "clang/Driver/Driver.h"
 #include "llvm/ADT/STLExtras.h"
 #include "llvm/ADT/SmallString.h"
+#include "llvm/Config/config.h"
```
Comment (nikic) [inline clang/tools/libclang/CIndexer.cpp:20]: This is a private LLVM header, see the comment at the top of the file. It cannot be included in other subprojects.

You either have to export this macro from llvm-config.h, or you have to add it to clang's config.h.

Comment (abhina-sree) [reply clang/tools/libclang/CIndexer.cpp:20]: Thanks, I created a fix for it here https://github.com/llvm/llvm-project/pull/75928

**Comment by chapuni** [comment]
Could we take other ways? I think they don't affect external API.

1. Detect stuff in `clang/Config/config.h`
2. Migrate the special logic to `llvm/Support`.

**Comment by abhina-sree** [comment]
> Could we take other ways? I think they don't affect external API.
> 
> 1. Detect stuff in `clang/Config/config.h`
> 2. Migrate the special logic to `llvm/Support`.

Thanks, I've created a patch here that detects the header in clang's config.h. Please let me know if this is a good approach

**Pull request merged**